torch/utils/data/datapipes/iter/combining.py

```diff
@@ -60,6 +62,39 @@ def __new__(cls, datapipe, instances):
         return [IterateBuffer(buffer) for i in range(instances)]
 
 
+@functional_datapipe('demux')
+class DemultiplexerIterDataPipe(IterDataPipe):
+
+    def __new__(cls, datapipe, instances, classifier_fn):
+        result = []
+        buffer = list(datapipe)
+
+        def filter_fn(classifier_fn, i, x):
+            return classifier_fn(x) == i
+        return [IterateBuffer(buffer).filter(functools.partial(filter_fn, classifier_fn, i)) for i in range(instances)]
+
+@functional_datapipe('mux')
+class MultiplexerIterDataPipe(IterDataPipe):
+
+    def __init__(self, *datapipes):
+        self.datapipes = datapipes
+
+    def __iter__(self):
+        iterators = [iter(x) for x in self.datapipes]
+        finished = {}
```

Comment:
Can we switch to `set()` here?

Reply:
I would prefer to keep it dict as eventually it might become dict with pipe->status (when we move to nonblocking world)